spotify_tensorflow/luigi/tfdv.py

```diff
@@ -57,10 +57,48 @@ def stats_file_name(self):
 
     def _get_output_args(self):
         input_targets = luigi.task.flatten(self.input())
-        assert len(input_targets) == 1, "Must provide 1 and only 1 input."
-        output_path = "{}/{}".format(get_uri(input_targets[0]),
-                                     self.stats_file_name)
-        return ["--output={}".format(output_path)]
+        if len(input_targets) != 1:
+            raise Exception("Must provide 1 and only 1 input.")
+
+        file_pattern_dict = self.file_pattern()
+        if len(file_pattern_dict) not in {0, 1}:
+            raise Exception("Only either 0 or 1 entry in file_pattern() is currently "
+                            "supported.")
+        file_pattern = None if len(file_pattern_dict) == 0 else list(file_pattern_dict.values())[0]
+        file_pattern_dir = TFDVGenerateStatsTask._get_dir_from_file_pattern(file_pattern)
+
+        base_uri = get_uri(input_targets[0])
+        stats_path = "{}/{}{}".format(base_uri.rstrip("/"), file_pattern_dir,
+                                      self.stats_file_name)
+
+        return ["--output={}".format(stats_path)]
+
+    @staticmethod
+    def _get_dir_from_file_pattern(file_pattern):
+        """
+        `file_pattern` may contain some directories. We want to get the directory
+        which closest to the actual data.
+
+        For example:
+        ```
+        "part-*" -> ""
+        "some/dir/part-*" -> "/some/dir/"
+        "some/dir*/file.txt" -> Exception
+        ```
+        """
+        if file_pattern is None:
+            return ""
+
+        file_pattern_dir = ""
+        if "/" in file_pattern:
```

Comment:
I guess this will always work for GCS paths, but it might be nice for this to work locally too. Could we check the scheme and if local, use os.path.sep instead?